server/index.ts

```diff
@@ -26,32 +29,76 @@ app.disable('x-powered-by')
 
 // Remix fingerprints its assets so we can cache forever.
 app.use(
-	'/build',
-	express.static('public/build', { immutable: true, maxAge: '1y' }),
+  '/build',
+  express.static('public/build', { immutable: true, maxAge: '1y' }),
 )
-
 // Everything else (like favicon.ico) is cached for an hour. You may want to be
 // more aggressive with this caching.
 app.use(express.static('public', { maxAge: '1h' }))
 
-morgan.token('url', (req, res) => decodeURIComponent(req.url ?? ''))
 app.use(morgan('tiny'))
 
+morgan.token('url', (req, res) => decodeURIComponent(req.url ?? ''))
+
+app.use((_, res, next) => {
+  res.locals.cspNonce = crypto.randomBytes(16).toString('hex')
+  next()
+})
+
+app.use(
+  helmet({
+    crossOriginEmbedderPolicy: false,
+    contentSecurityPolicy: {
+      directives: {
+        'connect-src': MODE === 'development' ? ['ws:', "'self'"] : null,
+        'font-src': ["'self'"],
+        'frame-src': [
+          "'self'",
+        ],
+        'img-src': [
+          "'self'",
+          'data:',
+          'user-images.githubusercontent.com',
+          'raw.githubusercontent.com',
+          ...(MODE === 'development' ? ['cloudflare-ipfs.com'] : []),
```

Comment:
I think I'd like to move the images to be served by the app server instead so we don't have to include this in the csp.